evm/src/cpu/kernel/tests/sha2.rs

```diff
@@ -0,0 +1,54 @@
+use std::str::FromStr;
+
+use anyhow::Result;
+use ascii::AsciiStr;
+use ethereum_types::U256;
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
+use sha2::{Digest, Sha256};
+
+use crate::cpu::kernel::aggregator::combined_kernel;
+use crate::cpu::kernel::interpreter::run;
+
+#[test]
+fn test_sha2() -> Result<()> {
+    let kernel = combined_kernel();
+    let sha2 = kernel.global_labels["sha2"];
+
+    let mut rng = thread_rng();
+
+    let num_bytes = rng.gen_range(1..10000);
+    let message: String = rng
+        .sample_iter(&Alphanumeric)
+        .take(num_bytes)
+        .map(char::from)
+        .collect();
```

Comment:
nit: I don't see why we should only test alphanumeric strings. We could just have `message` be random bytes:
```rust
let message: Vec<u8> = (0..num_bytes).map(|_| rng.gen()).collect();
```
That way we can also remove the `ascii` dependency.